tools/ci_connector_ops/ci_connector_ops/pipelines/tests/java_connectors.py

```diff
@@ -250,28 +222,28 @@ class IntegrationTestJava(GradleTask):
     title = "Integration tests"
     gradle_task_name = "integrationTestJava"
 
-    async def _load_normalization_image(self, normalization_tar_file: File):
+    async def _load_normalization_image(self, normalization_image_ref: str):
         normalization_image_tag = f"{self.context.connector.normalization_repository}:dev"
-        self.context.logger.info("Load the normalization image to the docker host.")
-        await environments.load_image_to_docker_host(
-            self.context, normalization_tar_file, normalization_image_tag, docker_service_name=self.docker_service_name
+        self.context.logger.info("Pull the normalization image to the docker host.")
+        await environments.pull_image_to_docker_host(
+            self.context, normalization_image_ref, normalization_image_tag, docker_service_name=self.docker_service_name
         )
-        self.context.logger.info("Successfully loaded the normalization image to the docker host.")
+        self.context.logger.info("Successfully pulled the normalization image to the docker host.")
 
-    async def _load_connector_image(self, connector_tar_file: File):
+    async def _load_connector_image(self, connector_image_ref: str):
         connector_image_tag = f"airbyte/{self.context.connector.technical_name}:dev"
-        self.context.logger.info("Load the connector image to the docker host")
-        await environments.load_image_to_docker_host(
-            self.context, connector_tar_file, connector_image_tag, docker_service_name=self.docker_service_name
+        self.context.logger.info("Pull the connector image to the docker host")
+        await environments.pull_image_to_docker_host(
+            self.context, connector_image_ref, connector_image_tag, docker_service_name=self.docker_service_name
```

Comment:
If the disk size incurred by saving all the tar files was too large, will the same issue happen when pulling all the docker images to use? I know after running all the CAT tests locally by pulling the images (plus a few more images) I currently use 123.1GB of disk space on docker images

Reply:
The thing is that when you pull an image its content is repesented by buildkit layers that can be fat but there storage is optimized and can be shared between images with same layers. When using tar file we dump to disk these full images. So we store the image content two times: in buildkit layers and on disk with tar files.

Reply:
The same things happen (the buildkit sharing) when building with docker locally, right? I didn't pull all of the images from dockerhub, I built them all locally. Could we, instead of build + push + pull docker images, only build and use the local ones? I guess I'm not clear on why we went the .tar path in the first place

Reply:
Sounds like that'd be the `I'm all in for a sidecar local registry we could use to share images with acceptances / integration tests.` approach you mentioned below 👍🏻